packages/components/src/box-control/input-control.tsx

```diff
@@ -0,0 +1,197 @@
+/**
+ * WordPress dependencies
+ */
+import { useInstanceId } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Tooltip from '../tooltip';
+import { parseQuantityAndUnitFromRawValue } from '../unit-control/utils';
+import {
+	CUSTOM_VALUE_SETTINGS,
+	getAllowedSides,
+	getMergedValue,
+	isValueMixed,
+	isValuesDefined,
+	LABELS,
+} from './utils';
+import {
+	FlexedBoxControlIcon,
+	FlexedRangeControl,
+	InputWrapper,
+	StyledUnitControl,
+} from './styles/box-control-styles';
+import type { BoxControlInputControlProps, BoxControlValue } from './types';
+
+const noop = () => {};
+
+function getSidesToModify(
+	side: BoxControlInputControlProps[ 'side' ],
+	sides: BoxControlInputControlProps[ 'sides' ],
+	isAlt?: boolean
+) {
+	const allowedSides = getAllowedSides( sides );
+
+	let modifiedSides: ( keyof BoxControlValue )[] = [];
+	switch ( side ) {
+		case 'all':
+			modifiedSides = [ 'top', 'bottom', 'left', 'right' ];
+			break;
+		case 'horizontal':
+			modifiedSides = [ 'left', 'right' ];
+			break;
+		case 'vertical':
+			modifiedSides = [ 'top', 'bottom' ];
+			break;
+		default:
+			modifiedSides = [ side ];
+	}
+
+	if ( isAlt ) {
+		switch ( side ) {
+			case 'top':
+				modifiedSides.push( 'bottom' );
+				break;
+			case 'bottom':
+				modifiedSides.push( 'top' );
+				break;
+			case 'left':
+				modifiedSides.push( 'left' );
+				break;
+			case 'right':
+				modifiedSides.push( 'right' );
+				break;
+		}
+	}
+
+	return modifiedSides.filter( ( s ) => allowedSides.has( s ) );
+}
+
+export default function BoxInputControl( {
+	__next40pxDefaultSize,
+	onChange = noop,
+	onFocus = noop,
+	values,
+	selectedUnits,
+	setSelectedUnits,
+	sides,
+	side,
```

Comment:
I think it could be useful to document why we provide both `side` and `sides` props.